### PR TITLE
Adds more styling to markdown elements

### LINKED ISSUE #5715: Markdown H1 vs H2 vs H3 styling
## Checklist

- [x] I have updated to the latest available Home Assistant version.
- [x] I have cleared the cache of my browser.
- [x] I have tried a different browser to see if it is related to my browser.

## The problem

The second level header seems odd/weird when rendered. It is smaller than H1 & H3.

image

## Expected behavior

The styling of the level 2 header to be somewhere between levels 1 & 3.

## Environment

- Home Assistant release with the issue: 0.110.0dev0
- Last working Home Assistant release (if known): unknown
- Browser and browser version: Any modern browser.
- Operating system: MacOS

## State of relevant entities

```yaml

```

## Problem-relevant configuration

```yaml

```

## Javascript errors shown in your browser console/inspector

```txt

```

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

Adds more styling to markdown elements, specifically code blocks and inline code elements.
This adds a new ccs var `--markdown-code-background-color` with a healthy default (IMO), the reason for a new var, is that none of the existing looked good, the closest was `--secondary-background-color`, but that was too dark with the default theme.

This also adds styling to `h2` where it was missing.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #5715
- This PR is related to issue:
- Link to documentation pull request:

**before examples:**
image

image

**after examples:**
image

image

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by zsarnett** [comment]
Why not use `--card-background-color`? Same as the Markdown Card. Everything seems to look good there

File: hassio/src/addon-view/documentation/hassio-addon-documentation-tab.ts [hunk 1]
```diff
@@ -64,6 +64,39 @@ class HassioAddonDocumentationDashboard extends LitElement {
...
+          margin-bottom: 0;
+        }
+        ha-markdown a {
+          color: var(--primary-color);
+        }
+        ha-markdown img {
+          max-width: 100%;
+        }
+        ha-markdown code,
+        pre {
```
Comment (balloob) [inline hassio/src/addon-view/documentation/hassio-addon-documentation-tab.ts:90]: Is this default styling that we should extract out and be able to include in any component that renderrs markdown and wants to use this styling? 

Also, let's add some padding to the code block like GitHub does.

Comment (ludeeus) [reply hassio/src/addon-view/documentation/hassio-addon-documentation-tab.ts:90]: I think I added it to all that uses it now, but it should probably be extracted yes.
I'll do that 👍

File: src/resources/styles.ts [hunk 2]
```diff
@@ -254,3 +254,50 @@ export const haStyleDialog = css`
     color: var(--google-red-500);
   }
 `;
+
+export const haMarkdown = css`
```
Comment (bramkragten) [inline src/resources/styles.ts:258]: Can we add this to `ha-markdown`?

Comment (ludeeus) [reply src/resources/styles.ts:258]: We can

File: src/panels/lovelace/cards/hui-markdown-card.ts [hunk 3]
```diff
@@ -80,7 +80,8 @@ export class HuiMarkdownCard extends LitElement implements LovelaceCard {
       <ha-card .header="${this._config.title}">
         <ha-markdown
           breaks
-          class="markdown ${classMap({
+          class="${classMap({
+            markdown: true,
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-markdown-card.ts:84]: revert this change

File: src/panels/lovelace/cards/hui-markdown-card.ts [hunk 4]
```diff
@@ -161,28 +162,9 @@ export class HuiMarkdownCard extends LitElement implements LovelaceCard {
 
   static get styles(): CSSResult {
     return css`
-      ha-markdown {
-        display: block;
-        padding: 0 16px 16px;
-        -ms-user-select: initial;
-        -webkit-user-select: initial;
-        -moz-user-select: initial;
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-markdown-card.ts:169]: This should also stay

File: src/panels/lovelace/cards/hui-markdown-card.ts (hunk 3 above)
Comment (ludeeus) [reply src/panels/lovelace/cards/hui-markdown-card.ts:84]: Why?

File: src/panels/lovelace/cards/hui-markdown-card.ts (hunk 4 above)
Comment (ludeeus) [reply src/panels/lovelace/cards/hui-markdown-card.ts:169]: ok

File: src/panels/lovelace/cards/hui-markdown-card.ts (hunk 3 above)
Comment (bramkragten) [reply src/panels/lovelace/cards/hui-markdown-card.ts:84]: Because there is no need to add it to a `classMap` if you want the class to be there always.

Comment (ludeeus) [reply src/panels/lovelace/cards/hui-markdown-card.ts:84]: I thought it looked cleaner using it for everything when it's in use 🤷‍♂️ 
But you are the boss, so revert it is! 🎉

File: src/components/ha-markdown.ts [hunk 5]
```diff
@@ -16,6 +16,9 @@ class HaMarkdown extends UpdatingElement {
   }
 
   private async _render() {
+    this.style.display = "block";
+    this.style.padding = "0 16px 16px";
```
Comment (bramkragten) [inline src/components/ha-markdown.ts:20]: Do we always want this padding? 

Also you should not set it on the element itself, it will overrule styles set by its parents.

We should use `:host() {}` (but that requires shadowRoot)

I see this element doesn't have a `shadowRoot`, we should add that if we are going to add styles, otherwise the styles will not be contained within this element.

File: src/components/ha-markdown-element.ts [hunk 6]
```diff
@@ -0,0 +1,69 @@
+import { customElement, property, UpdatingElement } from "lit-element";
+import { fireEvent } from "../common/dom/fire_event";
+import { renderMarkdown } from "../resources/render-markdown";
+
+@customElement("ha-markdown-element")
+class HaMarkdownElement extends UpdatingElement {
+  @property() public content = "";
```
Comment (bramkragten) [inline src/components/ha-markdown-element.ts:7]:
```suggestion
  @property() public content?;
```

File: src/components/ha-markdown-element.ts [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+  @property({ type: Boolean }) public allowSvg = false;
+
+  @property({ type: Boolean }) public breaks = false;
+
+  protected update(changedProps) {
+    super.update(changedProps);
+    this._render();
```
Comment (bramkragten) [inline src/components/ha-markdown-element.ts:15]:
```suggestion
    if (this.content !== undefined) {
    	this._render();
    }
```

File: src/panels/lovelace/cards/hui-markdown-card.ts (hunk 3 above)
Comment (bramkragten) [reply src/panels/lovelace/cards/hui-markdown-card.ts]: Might be cleaner, but it is definitely more expensive.
